solr/core/src/java/org/apache/solr/cluster/placement/AttributeFetcher.java

```diff
@@ -70,6 +71,12 @@
    */
   AttributeFetcher requestNodeMetric(String metricName, NodeMetricRegistry registry);
 
+  /**
+   * Request collection-level metrics. To get the values use {@link AttributeValues#getCollectionMetrics(String)}.
+   * Note that this request will fetch information from nodes relevant to the collection
+   * replicas and not the ones specified in {@link #fetchFrom(Set)} (though they may overlap).
+   */
+  AttributeFetcher requestCollectionMetrics(SolrCollection solrCollection, Set<ReplicaMetric<?>> metricNames);
```

Comment:
`metricNames` is not a good name, there are not just names.

Reply:
Initially this was a String but then I realized we need more metadata, I'll fix it.